beacon-chain/blockchain/process_block.go

```diff
@@ -149,19 +149,38 @@ func (s *Service) onBlock(ctx context.Context, signed interfaces.ReadOnlySignedB
 	if err := s.savePostStateInfo(ctx, blockRoot, signed, postState); err != nil {
 		return err
 	}
+
+	// Forkchoice critical section, write lock is held
+	s.cfg.ForkChoiceStore.Lock()
 	if err := s.cfg.ForkChoiceStore.InsertNode(ctx, postState, blockRoot); err != nil {
+		s.cfg.ForkChoiceStore.Unlock()
 		return errors.Wrapf(err, "could not insert block %d to fork choice store", signed.Block().Slot())
 	}
-	if err := s.handleBlockAttestations(ctx, signed.Block(), postState); err != nil {
-		return errors.Wrap(err, "could not handle block's attestations")
-	}
-
 	s.InsertSlashingsToForkChoiceStore(ctx, signed.Block().Body().AttesterSlashings())
 	if isValidPayload {
 		if err := s.cfg.ForkChoiceStore.SetOptimisticToValid(ctx, blockRoot); err != nil {
+			s.cfg.ForkChoiceStore.Unlock()
 			return errors.Wrap(err, "could not set optimistic block to valid")
 		}
 	}
+	start := time.Now()
+	headRoot, err := s.cfg.ForkChoiceStore.Head(ctx)
+	if err != nil {
+		log.WithError(err).Warn("Could not update head")
+	}
+	newBlockHeadElapsedTime.Observe(float64(time.Since(start).Milliseconds()))
+
+	if err := s.handleBlockAttestations(ctx, signed.Block(), postState); err != nil {
+		log.WithError(err).Warn("could not handle block's attestations")
+	}
```

Comment:
It looks like you changed the ordering. `handleBlockAttestations` should be before `Head`. Do we not want to count the attestations in the block?